lib/ansible/parsing/dataloader.py

```diff
@@ -115,8 +115,8 @@ def list_directory(self, path):
 
     def is_executable(self, path):
         '''is the given path executable?'''
-        path = self.path_dwim(path)
-        return is_executable(path)
+        b_path = to_bytes(self.path_dwim(path), errors='surrogate_or_strict')
+        return is_executable(b_path)
```

Comment:
I think that is_executable should be doing the to_bytes.  The reason is that is_executable() takes a native string because it is module_utils code and then making sure that it is bytes before passing it onto stat().  Otherwise is_executable could traceback at another time.